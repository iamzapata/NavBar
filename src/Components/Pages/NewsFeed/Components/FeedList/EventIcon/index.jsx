import React from "react"

const NewsIcon = () => (
  <svg
    width="30"
    height="30"
    viewBox="0 0 30 30"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g id="Symbols" fill="none" fillRule="evenodd">
      <g transform="translate(-30.5 -30)" id="card-text-with-image">
        <circle id="Oval-2" fill="#EEE" cx="45.5" cy="45" r="15" />
        <g transform="translate(39.5 39)" id="icon-event">
          <g id="frames-24px">
            <rect id="Rectangle-path" width="12" height="12" />
          </g>
          <g
            id="Nova_line"
            transform="translate(.5 1)"
            fill="#919191"
            fillRule="nonzero"
          >
            <g id="Group">
              <path
                d="M10,1 L8.5,1 L8.5,0 L7.5,0 L7.5,3 L8.5,3 L8.5,2 L10,2 L10,9.5 L1,9.5 L1,2 L2,2 L2,1 L1,1 C0.448,1 0,1.448 0,2 L0,9.5 C0,10.052 0.448,10.5 1,10.5 L10,10.5 C10.552,10.5 11,10.052 11,9.5 L11,2 C11,1.448 10.552,1 10,1 Z"
                id="Shape"
              />
              <polygon
                id="Shape"
                points="3.5 2 7 2 7 1 3.5 1 3.5 0 2.5 0 2.5 3 3.5 3"
              />
              <rect id="Rectangle-path" x="2" y="4" width="2" height="2" />
              <rect id="Rectangle-path" x="4.5" y="4" width="2" height="2" />
              <rect id="Rectangle-path" x="7" y="4" width="2" height="2" />
              <rect id="Rectangle-path" x="2" y="6.5" width="2" height="2" />
              <rect id="Rectangle-path" x="4.5" y="6.5" width="2" height="2" />
            </g>
          </g>
        </g>
      </g>
    </g>
  </svg>
)

export default NewsIcon
