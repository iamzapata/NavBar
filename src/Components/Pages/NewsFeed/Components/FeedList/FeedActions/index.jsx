import React from "react"
import "./FeedActions.scss"

const FeedActions = () => (
  <div className="FeedActions evel-right">
    <div className="level-item">
      <button className="button" type="button">
        <span className="icon">
          <i className="fa fa-ellipsis-h" aria-hidden="true" />
        </span>
      </button>
    </div>
  </div>
)

export default FeedActions
