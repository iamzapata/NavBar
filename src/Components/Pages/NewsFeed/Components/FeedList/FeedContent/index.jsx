import React from "react"
import { string } from "prop-types"

const createContentMarkup = content => ({ __html: content })

const FeedContent = ({ content }) => {
  return (
    <div
      className="FeedContent"
      dangerouslySetInnerHTML={createContentMarkup(content)}
    />
  )
}

FeedContent.propTypes = {
  content: string.isRequired
}

export default FeedContent
