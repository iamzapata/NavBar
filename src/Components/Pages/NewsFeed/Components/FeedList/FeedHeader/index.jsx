import React from "react"
import { distanceInWordsToNow } from "date-fns"
import "./FeedHeader.scss"

const FeedHeader = ({ feed: { title, author, updated_at } }) => (
  <div className="FeedHeader">
    <p className="title has-text-weight-normal is-3">{title}</p>
    <p className="subtitle is-6">
      By {author.name} &middot;
      {distanceInWordsToNow(updated_at)} ago
    </p>
  </div>
)

export default FeedHeader
