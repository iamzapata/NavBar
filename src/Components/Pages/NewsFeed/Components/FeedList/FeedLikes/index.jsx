import React from "react"
import { shape, number } from "prop-types"
import "./FeedLikes.scss"

const colleagues = likes => (likes > 1 ? "Colleagues" : "Colleague")

const FeedLikes = ({ feed: { likes } }) => (
  <span className="FeedLikes">
    <span className="Likes__Icon">
      <span className="icon">
        {likes > 0 && <i className="fa fa-heart" />}
        {likes === 0 && <i className="fa fa-heart-o" />}
      </span>
    </span>
    <span className="Likes__Count is-size-6">
      {likes > 0 && (
        <span>
          {`${likes} `}
          {colleagues(likes)}
          {" liked this"}
        </span>
      )}
    </span>
  </span>
)

FeedLikes.propTypes = {
  feed: shape({
    likes: number
  }).isRequired
}

export default FeedLikes
