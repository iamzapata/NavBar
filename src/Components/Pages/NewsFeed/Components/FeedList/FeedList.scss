@import "styles/variables.scss";

.FeedList {
  .FeedList__Header {
    color: #afafaf;
    font-size: 12px;
    margin: 10px 0;
  }

  .content > * {
    color: $color-gray-content-text;
  }
}
