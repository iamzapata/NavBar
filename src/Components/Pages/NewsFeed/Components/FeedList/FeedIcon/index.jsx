import React from "react"
import { shape, string } from "prop-types"
import ProductNewsIcon from "../ProductNewsIcon"
import EventIcon from "../EventIcon"
import "./FeedIcon.scss"

const FeedIcon = ({ feed: { type } }) => (
  <div className="FeedIcon level-left">
    <div className="level-item">
      {type === "Product News" ? <ProductNewsIcon /> : <EventIcon />}
      <span className="FeedIcon__Text">{type}</span>
    </div>
  </div>
)

FeedIcon.propTypes = {
  feed: shape({
    type: string.isRequired
  }).isRequired
}

export default FeedIcon
