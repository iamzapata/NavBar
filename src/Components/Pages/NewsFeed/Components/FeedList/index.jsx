import React from "react"
import { arrayOf, shape, number, string } from "prop-types"
import { doesArrayHaveData } from "utils/hasData"
import FeedHeader from "./FeedHeader"
import FeedIcon from "./FeedIcon"
import FeedActions from "./FeedActions"
import FeedLikes from "./FeedLikes"
import FeedContent from "./FeedContent"
import "./FeedList.scss"

const FeedList = ({ feed }) => (
  <div className="FeedList column is-7">
    <p className="FeedList__Header">LATEST</p>
    <div className="FeedList__Items">
      {doesArrayHaveData(feed) &&
        feed.map(f => (
          <div className="content" key={f.id}>
            <div className="box">
              <div className="level">
                <FeedIcon feed={f} />
                <FeedActions />
              </div>
              <FeedHeader feed={f} />
              <FeedContent content={f.content} />
              <hr />
              {f.media ? <img src={f.media} alt="Feed" /> : null}
              <FeedLikes feed={f} />
            </div>
          </div>
        ))}
    </div>
  </div>
)

FeedList.propTypes = {
  feed: arrayOf(
    shape({
      id: number.isRequired,
      type: string.isRequired,
      title: string.isRequired,
      likes: number.isRequired,
      content: string.isRequired,
      updated_at: string.isRequired
    })
  ).isRequired
}

export default FeedList
