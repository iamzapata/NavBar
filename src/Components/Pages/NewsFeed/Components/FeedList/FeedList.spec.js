import React from "react"
import { render, cleanup } from "react-testing-library"
import FeedList from "./index"

describe("<FeedList />", () => {
  afterEach(cleanup)
  it("Should render self", () => {
    const { container } = render(<FeedList feed={[]} />)
    expect(container.firstChild).toBeTruthy()
    expect(container.firstChild).toMatchSnapshot()
  })
})
