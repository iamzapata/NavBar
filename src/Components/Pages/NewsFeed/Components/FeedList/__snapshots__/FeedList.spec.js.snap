// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedList /> Should render self 1`] = `
<div
  class="FeedList column is-7"
>
  <p
    class="FeedList__Header"
  >
    LATEST
  </p>
  <div
    class="FeedList__Items"
  />
</div>
`;
